conformance/lang/expressions/error_constructor.balt

```diff
@@ -0,0 +1,526 @@
+    E1|E2 err2 = error("msg", code = x); // @error cannot infer type of the error from '(E1|E2)'
+}
+
+Test-Case: output
+Description: Test 7 for error constructor when the contextually expected type is a union.
+Labels: error-constructor-expr, module-type-defn, record-type, error-type, distinct-type, union-type, string, is-expr
+
+type E1 distinct error<record {int code;}>;
+
+type E2 distinct error<record {byte code;}>;
+
+function init() {
+    int x = 100;
+    E1|E2 err = error("msg", code = x);
+
+    io:println(err is E1); // @output true
+    io:println(err is E2); // @output false
+}
+
+Test-Case: output
+Description: Test type-ids of the constructed error value are those induced by applicable contextually expected type.
+Labels: error-constructor-expr, module-type-defn, record-type, error-type, distinct-type, intersection-type, string,
+        is-expr
+
+type E1 distinct error<record {int code;}>;
+
+type E2 distinct error<record {int no;}>;
+
+type E3 error<record {int code; int no;}>;
+
+type E E1 & E2;
+
+function init() {
+    E1 err1 = error("message", code = 100);
+    error err2 = err1;
+    io:println(err2 is E1); // @output true
+
+    E err3 = error("message", code = 100, no = 200);
+    E3 err4 = err3;
+    io:println(err4 is E); // @output true
+}
+
+Test-Case: output
+Description: Test nested error constructor.
+Labels: error-constructor-expr, module-type-defn, record-type, error-type, string, method-call-expr, if-else-stmt,
+        optional-type, is-expr, error:message, error:cause, error:detail
+
+type E error<record {|string message?; boolean fatal?;|}>;
+
+function init() {
+    error err1 = error("message 1", error E("message 2", message = "ERROR!", fatal = true));
+    io:println(err1.message()); // @output message 1
+    io:println(err1.detail()); // @output {}
+
+    error? cause1 = err1.cause();
+    if cause1 is error {
+        io:println(cause1.message()); // @output message 2
+        io:println(cause1.detail()); // @output {"message":"ERROR!","fatal":true}
+    }
+
+    error err2 = error("message 1", error E("message 2", error("message 3", error("message 4", fatal = false)), message = "ERROR!", fatal = true), errorCode = "BCE001");
+    io:println(err2.message()); // @output message 1
+    io:println(err2.detail()); // @output {"errorCode":"BCE001"}
+
+    error? cause2 = err2.cause();
+    if cause2 is error {
+        io:println(cause2.message()); // @output message 2
+        io:println(cause2.detail()); // @output {"message":"ERROR!","fatal":true}
+
+        error? cause3 = cause2.cause();
+        if cause3 is error {
+            io:println(cause3.message()); // @output message 3
+            io:println(cause3.detail()); // @output {}
+
+            error? cause4 = cause3.cause();
+            if cause4 is error {
+                io:println(cause4.message()); // @output message 4
+                io:println(cause4.detail()); // @output {"fatal":false}
+            }
+        }
+    }
+}
```

Comment:
To cover

> The stack trace in the constructed error value describes the execution stack at the point where the error constructor was evaluated.

this line, shall we print the stacktrace.

Reply:
A test was not added before because the stack track is going to be implementation-dependent. Anyway added a test.

Reply:
A test was not added before because the stack trace is going to be implementation-dependent. Anyway added a test.